lib/ts/utils/dateProvider/types.ts

```diff
@@ -14,9 +14,10 @@
  */
 
 export interface DateProviderInterface {
+    getThresholdInSeconds(): number;
     now(): number;
     setClientClockSkewInMillis(clockSkewInMillis: number): void;
     getClientClockSkewInMillis(): number;
 }
```

Comment:
we also need to allow the user to set their own threshold value. So make this a config please.

Reply:
Yeah we discussed this and decided against it as we thought, one can simply override the DateProvider if they want to do this.

But I have added it back as the overriding function doesn't get originaImplementation anymore.